src/components/toolbar/saved-notebooks-and-examples-subsection.jsx

```diff
@@ -1,47 +1,63 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
 import NotebookMenuItem from './notebook-menu-item'
 import NotebookMenuDivider from './notebook-menu-divider'
 import NotebookMenuHeader from './notebook-menu-header'
 import NotebookMenuSubsection from './notebook-menu-subsection'
 
+
 import iodideExampleTasks from '../../iodide-examples'
 import tasks, { getLocalStorageNotebook } from '../../task-definitions'
 import { stateFromJsmd } from '../../jsmd-tools'
 
-const AUTOSAVE = 'AUTOSAVE: '
-const autosave = Object.keys(localStorage).filter(n => n.includes(AUTOSAVE))[0] //
-const locallySaved = Object.keys(localStorage).filter(n => !n.includes(AUTOSAVE))
+class SavedNotebooksAndExamplesSubsection extends React.Component {
+  static propTypes = {
+    locallySaved: PropTypes.array,
+    autoSave: PropTypes.string,
+  }
 
-let autoSaveMenuItem
-let autoSave
-if (autosave !== undefined) {
-  autoSave = getLocalStorageNotebook(autosave)
-  autoSaveMenuItem = <NotebookMenuItem task={autoSave} key={autoSave.title} />
-}
+  getLocallySavedMenuItems() {
+    let locallySavedMenuItems
+    if (this.props.locallySaved) {
+      this.props.locallySaved.sort((a, b) => {
```

Comment:
can we move all this sorting stuff to the `getSavedProjects()` function in the notebook-reducer file? In general we're moving towards having our React components be as dumb as possible (makes them much easier to test), so ideally we'll just pass in the full sorted list as props to the component and render it. We should only ever have to sort this list when it changes, which we can do on load and then in subsequent reducer actions


see if you can get

move as much of `getLocallySavedMenuItems()` functionality out of the component as possible.